tuteria_application_test/users/views.py

```diff
@@ -45,3 +51,22 @@ class UserListView(LoginRequiredMixin, ListView):
     # These next two lines tell the view to index lookups by username
     slug_field = 'username'
     slug_url_kwarg = 'username'
+
+
+class UserApiView(APIView):
+    """API view for the user model."""
+    
+    serializer_class = UserSerializer
+    queryset = User.objects.all()
+
+    def get(self, request, *args, **kwargs):
+        serializer = UserSerializer(self.queryset.first())
```

Comment:
This is just wrong on many levels.  Fetching a record for any user would end up with the same result.
You do realize that you would be left to handle full implementations on your own and calling `self.queryset.first()` for every single item query is getting just the first result for every one that tries fetching a different user.

The problem isn't necessary with django rest framework. In fact there was nothing stopping you from using raw django view. The problem is the query which wrong.